src/main/java/Lemona.java

```diff
@@ -0,0 +1,93 @@
+import java.io.IOException;
+
+import oop.Ui;
+import oop.Parser;
+import oop.Storage;
+import oop.TaskList;
+
+import task.Todo;
+import task.Task;
+import task.Event;
+import task.Deadline;
+
+/**
+ * The main class representing the Lemona task manager application.
+ * Lemona allows users to manage their tasks by adding, listing, marking, and deleting tasks.
+ * It also provides the functionality to save tasks to a file and load tasks from a file.
+ */
+public class Lemona {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a Lemona object with the specified file path for task storage.
+     *
+     * @param filePath The file path to store tasks.
+     */
+    public Lemona(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the Lemona application.
+     * Displays a welcome message and continuously waits for user input.
+     * Processes user commands and performs corresponding actions.
+     * Exits the application when the "bye" command is entered.
+     */
+    public void run() {
+        ui.greet();
+        while (true) {
+            String input = ui.readInput();
+            String[] parsedInput = Parser.parse(input);
```

Comment:
According to the coding convention, you may rename `parsedInput` into `parsedInputs` (plural form), as it is a collection of Strings.